### PR TITLE
record the error of removing volumes when removing containers

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao mashimiao.fnst@cn.fujitsu.com

### CODE REVIEW

File: daemon/delete.go [hunk 1]
```diff
@@ -47,15 +47,16 @@ func (daemon *Daemon) ContainerRm(name string, config *ContainerRmConfig) error
 				logrus.Debugf("Could not update network to remove link %s: %v", n, err)
 			}
 		}
-
-		return nil
```
Comment (runcom) [inline daemon/delete.go]: not sure about this, `docker rm --link` is about just removing a link between two containers and this seems incorrect, I hope there's some test covering this also

**Comment by runcom** [comment]
https://docs.docker.com/reference/commandline/rm/

> This will remove the underlying link between /webapp and the /redis containers removing all network communication.

**Comment by Mashimiao** [comment]
Hi @runcom.  So this option is a specifial feature?

**Comment by runcom** [comment]
@Mashimiao this is how it should work, going backwards in history that `return nil` is always there as well

**Comment by Mashimiao** [comment]
OK. I will change this patch and title

**Comment by calavera** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**